src/v/storage/disk_log_impl.h

```diff
@@ -315,7 +315,7 @@ class disk_log_impl final : public log {
     model::offset _start_offset;
     // Used to serialize updates to _start_offset. See the update_start_offset
     // method.
-    mutex _start_offset_lock;
+    mutex _start_offset_lock{"disk_log_impl::start_offset_lock"};
```

Comment:
Is there any reasoning why some names have the class name in them while others don't?

Reply:
Well I sort of evolved my strategy over time. My overall thought was "it doesn't matter too much as these are going to be searched", but what I settled on was mostly: use `class name::member name` except in places where a class just had some single general mutex in which case I sometimes just used the class name. Also strip the leading `_`. Also remove "mutex" or "lock" for things named like `foo_mutex` or `foo_lock`.

I think most of the cases that "don't have class name" are actually cases where I only used the class name, and not the member name?

I'm fine to do anything really to get this in: if you suggest a strategy I will redo the change following that strategy.

Reply:
I also toyed with using `std::source_location` to keep the default constructor around and name the mutex based on say dir/file.cc:LINE but that seems mostly worse I think (if you're willing to do the work to rename them all).

Reply:
Yeah, I also immediately thought source_location, but then, there's a lot more useful info that could be added. I wouldn't block an improvement in the pursuit of the perfect bike-shed, though.

Reply:
Happy to merge as is.